src/Tickets/Seating/Service/Seat_Types.php

```diff
@@ -263,6 +266,152 @@ public function update_tickets_capacity( array $updates ): int {
 		return $total_updated;
 	}
 
+	/**
+	 * Updates the tickets moving to the with the $new_seat_type_id with the calculated stock and capacity.
+	 *
+	 * @since TBD
+	 *
+	 * @param string     $new_seat_type_id    The new seat type ID.
+	 * @param int        $new_capacity        The new capacity.
+	 * @param array<int> $original_ticket_ids The original ticket IDs that belonged to the new seat type prior this update.
+	 *
+	 * @return int The number of tickets updated.
+	 */
+	public function update_tickets_with_calculated_stock_and_capacity( string $new_seat_type_id, int $new_capacity, array $original_ticket_ids ): int {
+		if ( empty( $new_seat_type_id ) || empty( $new_capacity ) ) {
+			return 0;
+		}
+
+		$total_updated = 0;
+
+		/** @var Tribe__Tickets__Tickets_Handler $tickets_handler */
+		$tickets_handler   = tribe( 'tickets.handler' );
+		$capacity_meta_key = $tickets_handler->key_capacity;
+
+		$events_primary = [];
+
+		remove_action( 'updated_postmeta', [ tribe( Commerce_Controller::class ), 'sync_seated_tickets_stock' ] );
+
+		foreach (
+			tribe_tickets()
+				->where( 'meta_equals', Meta::META_KEY_SEAT_TYPE, $new_seat_type_id )
+				->not_in( $original_ticket_ids )
```

Comment:
This method should bail on empty `$original_ticket_ids` since it will build a query with an invalid `IN` clause in that case.

Reply:
Im confident it should not bail, but rather account for that. Something like `! empty( $original_ticket_ids ) ? $original_ticket_ids : 0` but ill write a test case about this scenario which is not being covered yet to make sure.

Reply:
The added test case confirms that we shouldn't bail but rather account for that case. Amazing catch by the way!